clients/java/src/main/java/io/zeebe/client/impl/ZeebeClientCloudBuilderImpl.java

```diff
@@ -0,0 +1,196 @@
+    return this;
+  }
+
+  @Override
+  public ZeebeClientBuilder brokerContactPoint(final String contactPoint) {
+    return innerBuilder.gatewayAddress(contactPoint);
+  }
+
+  @Override
+  public ZeebeClientBuilder gatewayAddress(final String gatewayAddress) {
+    return innerBuilder.gatewayAddress(gatewayAddress);
+  }
+
+  @Override
+  public ZeebeClientBuilder usePlaintext() {
+    return innerBuilder.usePlaintext();
+  }
+
+  @Override
+  public ZeebeClientBuilder credentialsProvider(final CredentialsProvider credentialsProvider) {
+    return innerBuilder.credentialsProvider(credentialsProvider);
+  }
+
+  @Override
+  public ZeebeClientBuilder withProperties(final Properties properties) {
+    return innerBuilder.withProperties(properties);
+  }
+
+  @Override
+  public ZeebeClientBuilder defaultJobWorkerMaxJobsActive(final int maxJobsActive) {
+    return innerBuilder.defaultJobWorkerMaxJobsActive(maxJobsActive);
+  }
+
+  @Override
+  public ZeebeClientBuilder numJobWorkerExecutionThreads(final int numThreads) {
+    return innerBuilder.numJobWorkerExecutionThreads(numThreads);
+  }
+
+  @Override
+  public ZeebeClientBuilder defaultJobWorkerName(final String workerName) {
+    return innerBuilder.defaultJobWorkerName(workerName);
+  }
+
+  @Override
+  public ZeebeClientBuilder defaultJobTimeout(final Duration timeout) {
+    return innerBuilder.defaultJobTimeout(timeout);
+  }
+
+  @Override
+  public ZeebeClientBuilder defaultJobPollInterval(final Duration pollInterval) {
+    return innerBuilder.defaultJobPollInterval(pollInterval);
+  }
+
+  @Override
+  public ZeebeClientBuilder defaultMessageTimeToLive(final Duration timeToLive) {
+    return innerBuilder.defaultMessageTimeToLive(timeToLive);
+  }
+
+  @Override
+  public ZeebeClientBuilder defaultRequestTimeout(final Duration requestTimeout) {
+    return innerBuilder.defaultRequestTimeout(requestTimeout);
+  }
+
+  @Override
+  public ZeebeClientBuilder caCertificatePath(final String certificatePath) {
+    return innerBuilder.caCertificatePath(certificatePath);
+  }
+
+  @Override
+  public ZeebeClientBuilder keepAlive(final Duration keepAlive) {
+    return innerBuilder.keepAlive(keepAlive);
+  }
+
+  @Override
+  public ZeebeClientBuilder withInterceptors(final ClientInterceptor... interceptor) {
+    return innerBuilder.withInterceptors(interceptor);
+  }
+
+  @Override
+  public ZeebeClientBuilder withJsonMapper(final JsonMapper jsonMapper) {
+    return innerBuilder.withJsonMapper(jsonMapper);
+  }
+
+  @Override
+  public ZeebeClient build() {
+    if (isNeedToSetCloudGatewayAddress()) {
+      ensureNotNull("cluster id", clusterId);
+      final String cloudGatewayAddress = String.format("%s.%s:443", clusterId, BASE_ADDRESS);
+      innerBuilder.gatewayAddress(cloudGatewayAddress);
+    } else {
+      Loggers.LOGGER.warn(
+          "The gateway address is overwritten. You are trying to connect *NOT* to the Camunda Cloud.");
+    }
```

Comment:
I think we can make it a bit more clear by extracting this into a separate private method called something like `determineGatewayAddress` that returns the gateway address to use, and then use that value on `innerBuilder.gatewayAddress(x)`. 

Something like:
```
innerBuilder.gatewayAddress(determineGatewayAddress());
```

You can do something similar for the credentials provider. That might make the logic more clear.

Reply:
Done